### ISSUE TITLE
RS1038 is being triggered for analyzer + codefixer project when only codefixers reference the workspace types

### ISSUE THREAD

**Issue by Meir017** [dotnet/roslyn-analyzers#7438]
### Analyzer

[RS1038](https://github.com/dotnet/roslyn-analyzers/blob/main/docs/rules/RS1038.md): This compiler extension should not be implemented in an assembly containing a reference to Microsoft.CodeAnalysis.Workspaces. The Microsoft.CodeAnalysis.Workspaces assembly is not provided during command line compilation scenarios, so references to it could cause the compiler extension to behave unpredictably. 

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)

**Version**: 3.11.0

### Steps To Reproduce

```ps1
dotnet new classlib --framework netstandard2.0
dotnet add package Microsoft.CodeAnalysis.Analyzers --version 3.3.4
dotnet add package Microsoft.CodeAnalysis.CSharp.Workspaces
```

create a new file
```cs
using System.Collections.Immutable;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.Diagnostics;

namespace DotnetAnalyzer;

[DiagnosticAnalyzer(LanguageNames.CSharp)]
public class TestDiagnosticAnalyzer : DiagnosticAnalyzer
{
    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => throw new NotImplementedException();

    public override void Initialize(AnalysisContext context)
    {
        context.EnableConcurrentExecution();
        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
    }
}
```

when running `dotnet build`, no warnning.

run:
```ps1
dotnet add package Microsoft.CodeAnalysis.Analyzers --version 3.11.0
```

when running `dotnet build`, warnning `warning RS1038`.

### Expected behavior

Only if any if the types exposed from the Workspace nuget packages are used from a `DiagnosticAnalyzer` then report the warning diagnostic.

### Actual behavior

Warning is always reported regarded if types from the Workspace nuget packages are used.

## Additional context

The reason for using the `Microsoft.CodeAnalysis.CSharp.Workspaces` package from a code analyzer project is that both the diagnostic-analyzer and the code-fix-provider live in the same project, and only the code-fix-provider uses types from Workspace nuget packages

**Comment by kimpenhaus** [comment]
@Meir017 did you found any other workaround other then suppress that warning/error?

**Comment by Meir017** [comment]
same warning reproduces with the latest version [3.12.0-beta1.25076.4](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers/3.12.0-beta1.25076.4) as well.

I noticed that 
```xml
<PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.13.0" />
```
has upgraded the version of the analyzers package in it's dependency list:
[Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers/) (>= 3.11.0)

**Comment by DmitryLukyanov** [comment]
I see the same behavior, any updates?
